packages/salesforcedx-vscode-core/src/deploys/toolingDeploy.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Connection } from '@salesforce/core';
+import * as fs from 'fs';
+import * as path from 'path';
+import { nls } from '../messages';
+import {
+  DeployStatusEnum,
+  supportedToolingTypes,
+  ToolingCreateResult,
+  ToolingRetrieveResult
+} from './index';
+// tslint:disable-next-line:no-var-requires
+const DOMParser = require('xmldom-sfdx-encoding').DOMParser;
+const CONTAINER_ASYNC_REQUEST = 'ContainerAsyncRequest';
+const METADATA_CONTAINER = 'MetadataContainer';
+
+export class ToolingDeploy {
+  public metadataType?: string;
+  public connection: Connection;
+  private apiVersion?: string;
+
+  public constructor(connection: Connection, apiVersion?: string) {
+    this.connection = connection;
+    if (apiVersion) this.apiVersion = apiVersion;
+  }
+
+  public async deploy(filePath: string) {
```

Comment:
FYI: I simplified things a bit since this is the first version of the deploy functionality.